code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -12,50 +12,52 @@
 	if (!check_and_use_plasma_owner())
 		return
 
+	//X = xeno user, A = target atom
+	var/list/turf/target_turfs = getline2(X, A, include_from_atom = FALSE)
+	var/len = LAZYLEN(target_turfs)
+	if(len > 3)
+		target_turfs = target_turfs.Copy(1, 4)
+
```

Comment:
I'd appreciate if someone could provide a more graceful solution here.

I'm trying to draw a line of turfs between the user and their target click but only take the first 3 turfs. `getline2()` doesn't have a limit param. So I just got all the turfs and discarded the ones I don't need. Feels wrong.

Reply:
This'd involve making a new proc that acts like `getline2()` but takes a numerical param and stops if the amt of turfs is >= the param